contrib/internet-urllib3.py

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env python
+
+import urllib3
+import certifi
+
+def isInternetConnected(url="www.ietf.org"):
+  result = False
+  http = urllib3.PoolManager(
+    cert_reqs='CERT_REQUIRED', # Force certificate check.
+    ca_certs=certifi.where(),  # Path to the Certifi bundle.
+  )
+  try:
+    r = http.request('HEAD', 'https://' + url)
+    result = True
+  except Exception as e:  # urllib3.exceptions.SSLError
+    result = False
```

Comment:
The technique is the same for https or IMAP secure connection - when you verify the remote server ssl cert is authentic, this proves the internet is connected.  A fake cert is obviously spoofed by a router captive portal, or other man-in-the-middle, so the internet is not connected.